shared/bytesutil/bytes.go

```diff
@@ -241,3 +243,82 @@ func PadTo(b []byte, size int) []byte {
 	}
 	return append(b, make([]byte, size-len(b))...)
 }
+
+// SetBit sets the index `i` of bitlist `b` to 1.
+// It grows and returns a longer bitlist with 1 set
+// if index `i` is out of range.
+func SetBit(b []byte, i int) []byte {
+	if i >= len(b)*8 {
+		h := (i + (8 - i%8)) / 8
+		b = append(b, make([]byte, h-len(b))...)
+	}
+
+	bit := uint8(1 << (i % 8))
+	b[i/8] |= bit
+	return b
+}
+
+// ClearBit clears the index `i` of bitlist `b`.
+// Returns the original bitlist if the index `i`
+// is out of range.
+func ClearBit(b []byte, i int) []byte {
+	if i >= len(b)*8 {
+		return b
+	}
+
+	bit := uint8(1 << (i % 8))
+	b[i/8] &^= bit
+	return b
+}
+
+// MakeEmptyBitfields returns an empty bitlist with
+// input size `i`.
+func MakeEmptyBitfields(i int) []byte {
```

Comment:
check naming consistency here, everything else is "bitlist" and this one "bitfield" in godocs